pages/api/index.ts

```diff
@@ -1,1632 +1,1704 @@
+                        return res.status(200).send({
+                            success: true,
+                            message: reason
+                        });
+                    }
+
+                    const label = await linear.issueLabel(removedLabelId);
+                    if (!label) {
+                        return res.status(403).send({
+                            success: false,
+                            message: "Could not find label."
+                        });
+                    }
+
+                    const removedLabelResponse = await got.delete(
+                        `${GITHUB.REPO_ENDPOINT}/${syncedIssue.GitHubRepo.repoName}/issues/${syncedIssue.githubIssueNumber}/labels/${label.name}`,
+                        {
+                            headers: {
+                                Authorization: githubAuthHeader,
+                                "User-Agent": userAgentHeader
+                            }
+                        }
+                    );
+
+                    if (removedLabelResponse.statusCode > 201) {
+                        console.log(`Could not remove label "${label.name}".`);
+                        return res.status(403).send({
+                            success: false,
+                            message: `Could not remove label "${label.name}".`
+                        });
+                    } else {
+                        console.log(
+                            `Removed label "${label.name}" from issue #${syncedIssue.githubIssueNumber}.`
+                        );
+                    }
+                } else if (data.labelIds.length > updatedFrom.labelIds.length) {
+                    const addedLabelId = data.labelIds.find(
+                        id => !updatedFrom.labelIds.includes(id)
+                    );
+
+                    const label = await linear.issueLabel(addedLabelId);
+                    if (!label) {
+                        return res.status(403).send({
+                            success: false,
+                            message: "Could not find label."
+                        });
+                    }
+
+                    const createdLabelResponse = await got.post(
+                        `https://api.github.com/repos/${repoFullName}/labels`,
+                        {
+                            headers: {
+                                Authorization: githubAuthHeader,
+                                "User-Agent": userAgentHeader
+                            },
+                            json: {
+                                name: label.name,
+                                color: label.color.replace("#", ""),
+                                description: "Created by Linear-GitHub Sync"
+                            }
+                        }
+                    );
+
+                    const createdLabelData = JSON.parse(
+                        createdLabelResponse.body
+                    );

```

Comment:
Got has an internal json parser, but because we use the response codes up, we need to parse it manually.